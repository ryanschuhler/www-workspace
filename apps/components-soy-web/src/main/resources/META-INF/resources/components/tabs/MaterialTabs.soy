{namespace MaterialTabs}

/**
 * Toolbar
 */
{template .render}
  {@param items: list<[
  active: bool,
  label: string,
  url: string
  ]>}

  <nav id="basic-tab-bar" class="mdc-tab-bar">
    {foreach $item in $items}
      {call ._tabItem}
        {param item: $item /}
      {/call}
    {/foreach}
    <span class="mdc-tab-bar__indicator"></span>
  </nav>
{/template}

/**
 * Toolbar item.
 */
{template ._tabItem}
  {@param item: [
  active: bool,
  label: string,
  url: string
  ]}

  <a
      class="mdc-tab {if $item.active}mdc-tab--active{/if}"
      href="{$item.url}"
  >{$item.label}</a>
{/template}