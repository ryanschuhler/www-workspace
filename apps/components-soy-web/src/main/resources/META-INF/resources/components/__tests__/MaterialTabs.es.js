'use strict';

import MaterialTabs from '../tabs/MaterialTabs.es';

const tabs = [
  {
    active: true,
    label: 'Current Tab',
    url: 'javascript:;'
  },
  {
    active: false,
    label: 'button 2',
    url: 'javascript:;'
  }
];

let component;

describe('MaterialTabs', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {};
  });

  it('should render', () => {
    component = new MaterialTabs({
      items: tabs
    });

    expect(component).toMatchSnapshot();
  });
});