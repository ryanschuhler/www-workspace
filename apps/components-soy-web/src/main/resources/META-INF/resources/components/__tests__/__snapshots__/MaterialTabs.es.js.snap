// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialTabs should render 1`] = `
<nav id="basic-tab-bar" class="mdc-tab-bar">
  <a class="mdc-tab mdc-tab--active" href="javascript:;">Current Tab</a>
  <a class="mdc-tab " href="javascript:;">button 2</a>
  <span class="mdc-tab-bar__indicator"></span>
</nav>
`;
